File: nan21.dnet.module.bd/nan21.dnet.module.bd.presenter/src/main/java/net/nan21/dnet/module/bd/attr/ds/model/AttributeSetAttributeDs.java
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */
package net.nan21.dnet.module.bd.attr.ds.model;

import net.nan21.dnet.core.presenter.model.base.AbstractAuditableDs;

import net.nan21.dnet.module.bd.attr.domain.entity.AttributeSetAttribute;
import net.nan21.dnet.core.api.annotation.Ds;
import net.nan21.dnet.core.api.annotation.DsField;

@Ds(entity = AttributeSetAttribute.class)
public class AttributeSetAttributeDs extends
        AbstractAuditableDs<AttributeSetAttribute> {

    public static final String fSETID = "setId";
    public static final String fSET = "set";
    public static final String fATTRIBUTEID = "attributeId";
    public static final String fATTRIBUTE = "attribute";
    public static final String fSEQUENCENO = "sequenceNo";
    public static final String fINDESCRIPTION = "inDescription";

    @DsField(join = "left", path = "attributeSet.id")
    private Long setId;

    @DsField(join = "left", path = "attributeSet.name")
    private String set;

    @DsField(join = "left", path = "attribute.id")
    private Long attributeId;

    @DsField(join = "left", path = "attribute.name")
    private String attribute;

    @DsField()
    private Integer sequenceNo;

    @DsField()
    private Boolean inDescription;

    public AttributeSetAttributeDs() {
        super();
    }

    public AttributeSetAttributeDs(AttributeSetAttribute e) {
        super(e);
    }

    public Long getSetId() {
        return this.setId;
    }

    public void setSetId(Long setId) {
        this.setId = setId;
    }

    public String getSet() {
        return this.set;
    }

    public void setSet(String set) {
        this.set = set;
    }

    public Long getAttributeId() {
        return this.attributeId;
    }

    public void setAttributeId(Long attributeId) {
        this.attributeId = attributeId;
    }

    public String getAttribute() {
        return this.attribute;
    }

    public void setAttribute(String attribute) {
        this.attribute = attribute;
    }

    public Integer getSequenceNo() {
        return this.sequenceNo;
    }

    public void setSequenceNo(Integer sequenceNo) {
        this.sequenceNo = sequenceNo;
    }

    public Boolean getInDescription() {
        return this.inDescription;
    }

    public void setInDescription(Boolean inDescription) {
        this.inDescription = inDescription;
    }

}

File: nan21.dnet.module.bd/nan21.dnet.module.bd.ui.extjs/src/main/webapp/src/ds/AttributeSetAttributeDs.js
/*
 * DNet eBusiness Suite
 * Copyright: 2008-2012 Nan21 Electronics SRL. All rights reserved.
 * Use is subject to license terms. 
 */   	 
Ext.define("net.nan21.dnet.module.bd.attr.ds.model.AttributeSetAttributeDs" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"clientId", type:"int", useNull:true},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"version", type:"int", useNull:true},
		{name:"entityFQN", type:"string"},
		{name:"setId", type:"int", useNull:true},
		{name:"set", type:"string"},
		{name:"attributeId", type:"int", useNull:true},
		{name:"attribute", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"inDescription", type:"boolean"}
	]
});

Ext.define("net.nan21.dnet.module.bd.attr.ds.model.AttributeSetAttributeDsFilter" ,{
	extend: 'Ext.data.Model',
	fields: [ 
		{name:"id", type:"int", useNull:true},
		{name:"uuid", type:"string"},
		{name:"createdAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt", type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_From",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"modifiedAt_To",type:"date", dateFormat:Dnet.MODEL_DATE_FORMAT},
		{name:"createdBy", type:"string"},
		{name:"modifiedBy", type:"string"},
		{name:"entityFQN", type:"string"},
		{name:"setId", type:"int", useNull:true},
		{name:"setId_From",type:"int", useNull:true},
		{name:"setId_To",type:"int", useNull:true},
		{name:"set", type:"string"},
		{name:"attributeId", type:"int", useNull:true},
		{name:"attributeId_From",type:"int", useNull:true},
		{name:"attributeId_To",type:"int", useNull:true},
		{name:"attribute", type:"string"},
		{name:"sequenceNo", type:"int", useNull:true},
		{name:"sequenceNo_From",type:"int", useNull:true},
		{name:"sequenceNo_To",type:"int", useNull:true},
		{name:"inDescription", type:"boolean", useNull:true}
	] 
});
	
Ext.define("net.nan21.dnet.module.bd.attr.ds.param.AttributeSetAttributeDsParam", {
	extend: 'Ext.data.Model',
	fields: [ 
	]
}); 
 